### DISCUSSION TITLE
Enable to add to Add AWS session token while trying to connect with amazon s3

### DISCUSSION THREAD

**Discussion by Moeez-nclouds** [iterate-ch/cyberduck#14928] [Q&A] [answered]
Hi everyone,
I hope this message finds you well.

I am currently attempting to establish a connection with Amazon S3 using Cyberduck while utilizing temporary AWS credentials. However, I have encountered an obstacle as I cannot locate the designated field to input the AWS session token. According to some resources, the Session Token field in Cyberduck might not be readily visible by default. To make it visible, one is advised to navigate to Preferences > Connections > Amazon S3 and select the option to Show Session Token. Nevertheless, I have followed these instructions diligently, yet the checkbox remains elusive. I would sincerely appreciate any assistance or guidance you can provide on this matter.
Thank you
![image](https://github.com/iterate-ch/cyberduck/assets/102989700/0ac0811e-674d-4a36-a7d4-bb0ae4d946ea)

![image](https://github.com/iterate-ch/cyberduck/assets/102989700/4fa4469c-f459-41c0-aa10-41ac4968ed63)

**Comment by dkocher** [comment]
There is support connecting to S3 with temporary credentials including a session token by using dedicated connection profiles.
- [Connecting with Temporary Access Credentials (Token) from EC2](https://docs.cyberduck.io/protocols/s3/#id13)
- [Connecting Using Credentials from AWS Command Line Interface](https://docs.cyberduck.io/protocols/s3/#id14)
- [Connecting Using AssumeRole from AWS Security Token Service (STS)](https://docs.cyberduck.io/protocols/s3/#id15)- [AWS Single Sign-On](https://docs.cyberduck.io/protocols/s3/#aws-single-sign-on)

There is work in progress [^1]  to facilitate the setup of connections using `AssumeRole` APIs of STS in particular when used with an identity provider (OAuth).

[^1]: #13804

**Comment by Moeez-nclouds** [comment]
Hi @dkocher ,
I hope you are well,
Thank you for your response, I tried to download the profile form the links that you mentioned but i'm getting this error. Can you please guide me regarding this,
Thanks
![Screenshot from 2023-07-25 22-03-51](https://github.com/iterate-ch/cyberduck/assets/102989700/a8283220-c0b2-47cd-b813-1a813304023f)

 
![image](https://github.com/iterate-ch/cyberduck/assets/102989700/7280756f-1bd8-4ebd-a105-39fec7eb6e54)

**Reply by Xarno** [reply, marked as answer]
That is not an error. You need to save this XML file. Your Browser thought it could be helpful by showing you the contents, but you need the file „someName.cyberduckprofile“ on your harddrive. Then doubleclick it.

**Comment by dkocher** [comment]
Currently because of a regression [^1] the profiles are not working as expected in the current release [8.6.2](https://trac.cyberduck.io/milestone/8.6.2).

[^1]: #14970
